scripts/aws_replicate.py

```diff
@@ -62,122 +76,230 @@ def __init__(self, global_config, manifest_file, thread_num, job_name, bucket=No
 
     def prepare_data(self):
         """
-        Read data file info from manifest and organize them into  groups.
-        Each group contains files should be copied to same bucket
-        The groups will be push to the queue
+        Read data file info from manifest and organize them into groups.
+        Each group contains files which should be copied to the same bucket
+        The groups will be push to the queue consumed by threads
         """
         if self.manifest_file.startswith("s3://"):
-            submitting_files = get_fileinfo_list_from_s3_manifest(self.manifest_file)
+            copying_files = get_fileinfo_list_from_s3_manifest(self.manifest_file)
         else:
-            submitting_files = get_fileinfo_list_from_csv_manifest(self.manifest_file)
+            copying_files = get_fileinfo_list_from_csv_manifest(self.manifest_file)
 
-        self.total_files = len(submitting_files)
+        self.total_files = len(copying_files)
 
+        chunk_size = self.global_config.get("chunk_size", 1)
         tasks = []
-        file_grps = exec_files_grouping(submitting_files)
 
-        for _, files in file_grps.iteritems():
-            chunk_size = self.global_config.get("chunk_size", 1)
-            idx = 0
-            while idx < len(files):
-                tasks.append(files[idx : idx + chunk_size])
-                idx = idx + chunk_size
+        idx = 0
+        while idx < len(copying_files):
```

Comment:
you can use `for idx in range(0, len(copying_files), chunk_size)`

Reply:
fixed in bf3aa835ec7175fe766666aed7f55b31dd5f725c